UnityProject/Assets/Scripts/Items/Others/HolyBook.cs

```diff
@@ -0,0 +1,147 @@
+﻿﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Mirror;
+using Random = System.Random;
+/// <summary>
+/// Component which allows this object to heal or cause brain damage if used by the Chaplain.
+/// </summary>
+public class HolyBook: MonoBehaviour, IPredictedCheckedInteractable<PositionalHandApply>
+{
+	private static readonly StandardProgressActionConfig ProgressConfig =
+		new StandardProgressActionConfig(StandardProgressActionType.SelfHeal);
+
+	private static Random rnd = new Random();
+
+	//The amount a single thwack heals or damages.
+	public int healthModifier = 10;
+
+	//Using the holy book is considered a melee attack.
+	public void ClientPredictInteraction(PositionalHandApply interaction)
+	{
+		//start clientside melee cooldown so we don't try to spam melee
+		//requests to server
+		Cooldowns.TryStartClient(interaction, CommonCooldowns.Instance.Melee);
+	}
+
+	//no rollback logic
+	public void ServerRollbackClient(PositionalHandApply interaction) { }
+	public bool WillInteract(PositionalHandApply interaction, NetworkSide side)
+	{
+		if (!DefaultWillInteract.Default(interaction, side)) return false;
+
+		//Occurs only if applied on the head.
+		if (interaction.TargetBodyPart != BodyPartType.Head) return false;
+
+		//Only the Chaplain can use the holy book.
+		if (PlayerList.Instance.Get(interaction.Performer).Job != JobType.CHAPLAIN) return false;
```

Comment:
These Validations should be in the server not the client

Reply:
Moved these checks onto ServerPerformInteraction instead.